server/src/immich/api-v1/asset/dto/import-asset.dto.ts

```diff
@@ -0,0 +1,48 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { IsOptional, IsString, IsBoolean } from 'class-validator';
+import fs from 'fs/promises';
+import path from 'path';
+import crypto from 'crypto';
+import mime from 'mime-types';
+import { CreateAssetDto } from './create-asset.dto';
+
+export class ImportAssetDto extends CreateAssetDto {
+  @IsOptional()
+  @IsBoolean()
+  isReadOnly?: boolean = true;
+
+  @ApiProperty()
+  @IsString()
+  assetData!: string;
+
+  @ApiProperty()
+  @IsString()
+  @IsOptional()
+  livePhotoData?: string;
+
+  @ApiProperty()
+  @IsString()
+  @IsOptional()
+  sidecarData?: string;
+}
+
+export interface ImportFile {
+  mimeType: string;
+  checksum: Buffer;
+  originalPath: string;
+  originalName: string;
+}
```

Comment:
Is there not already an immich file interface we can use or that is already used in asset core create?

Reply:
I went with ImportFile for consistency in case there were any future cases, but that can always be modified later. Removed this in favor of UploadFile.